server/api.go

```diff
@@ -495,6 +505,116 @@ func (p *Plugin) getYourPrs(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func (p *Plugin) getPrsDetails(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	var prList []*PRDetails
+	json.NewDecoder(r.Body).Decode(&prList)
+
+	var prDetails []*PRDetails
+
+	ch := make(chan *PRDetails)
+	for _, pr := range prList {
+		go fetchPRDetails(ctx, githubClient, pr.URL, pr.Number, ch)
+	}
+
+	for range prList {
+		pr := <-ch
```

Comment:
@lieut-data Do you mind taking a look at the the flow of goroutines and channels used in this PR? I'm concerned about the risk of this line blocking indefinitely if a given goroutine does not successfully send to the channel.

Reply:
Happy to help! Taking a look now :)